driver/apdu/at.c

```diff
@@ -9,35 +9,170 @@
 #include <euicc/interface.h>
 #include <euicc/hexutil.h>
 
-static FILE *fuart;
+// Include necessary headers for serial communication
+#include <fcntl.h>
+#include <stdint.h>
+#include <termios.h>
+#include <sys/file.h>
+
+// Declare static variables
+static int fd = -1;
 static int logic_channel = 0;
 
+// Function to open AT device
+static int open_at_device(const char *device)
+{
+    fd = open(device, O_RDWR | O_NOCTTY);
+    if (fd == -1)
+    {
+        fprintf(stderr, "Failed to open device: %s\n", device);
```

Comment:
```suggestion
        fprintf(stderr, "Failed to open device: %s: %s\n", device, strerror(errno));
```
It may be better to print the exact reason at the same time. Also for error handle code below.